sched/wdog/wd_start.c

```diff
@@ -165,40 +247,20 @@ static inline void wd_expiration(void)
  *
  ****************************************************************************/
 
-int wd_start(FAR struct wdog_s *wdog, sclock_t delay,
-             wdentry_t wdentry, wdparm_t arg)
+int wd_start_absolute(FAR struct wdog_s *wdog, clock_t ticks,
+                      wdentry_t wdentry, wdparm_t arg)
 {
-  FAR struct wdog_s *curr;
-  FAR struct wdog_s *prev;
-  FAR struct wdog_s *next;
-  sclock_t now;
   irqstate_t flags;
+  bool reassess = false;
 
   /* Verify the wdog and setup parameters */
 
-  if (wdog == NULL || wdentry == NULL || delay < 0)
+  if (wdog == NULL || wdentry == NULL)
     {
       return -EINVAL;
     }
 
-  /* Check if the watchdog has been started. If so, stop it.
-   * NOTE:  There is a race condition here... the caller may receive
-   * the watchdog between the time that wd_start is called and
-   * the critical section is established.
-   */
-
-  flags = enter_critical_section();
-  if (WDOG_ISACTIVE(wdog))
-    {
-      wd_cancel(wdog);
-    }
-
-  /* Save the data in the watchdog structure */
-
-  wdog->func = wdentry;         /* Function to execute when delay expires */
-  up_getpicbase(&wdog->picbase);
-  wdog->arg = arg;
-
+#ifdef CONFIG_WDOG_INTRESERVE
```

Comment:
This configuration should be deleted. If it is turned off, wdt may not be woken up.

Reply:
No, if don't set CONFIG_WDOG_INTRESERVE, not exactly like yours.

Should we consider just remove the ticks++ from code ?

Now we have CONFIG_USEC_PER_TICK, and for our timer system, all the calculation used 'tick'.
And all the timespec should change to 'tick' before use wd_start(), so USEC2TICK() can NOT be avoided.

Then there must be an 'less then one tick' loss.

Current resolution is ticks++ anyway when wd_start(). But this will caused time expired more a tick.
Other resolution:
I think we can let the XSEC2TICK() calculation use roundup